packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -365,12 +365,26 @@ export class Key extends KeyBase {
   constructor(scope: Construct, id: string, props: KeyProps = {}) {
     super(scope, id);
 
-    this.policy = props.policy || new iam.PolicyDocument();
-    this.trustAccountIdentities = props.trustAccountIdentities || false;
-    if (this.trustAccountIdentities) {
-      this.allowAccountIdentitiesToControl();
+    const defaultKeyPoliciesEnabled = FeatureFlags.of(this).isEnabled(cxapi.KMS_DEFAULT_KEY_POLICIES);
+
+    this.policy = props.policy ?? new iam.PolicyDocument();
+    if (defaultKeyPoliciesEnabled) {
+      if (props.trustAccountIdentities === false) {
+        Annotations.of(this).addWarning('`trustAccountIdentities` has no impact if the @aws-cdk/aws-kms:defaultKeyPolicies feature flag is set');
```

Comment:
I am confused, if the feature flag is set it just means that by default `trustAccountIdentities` is enabled, but what if I want to disable it. How do I do that?

Reply:
Why not just `trustAccountIdentities: false` as a way to disable this behavior?

Reply:
> Why not just `trustAccountIdentities: false` as a way to disable this behavior?

What are we trying to achieve by making it easy to disable this behavior? Asked another way, what value does the `trustAccountIdentities: false` behavior provide? With `trustAccountIdentities: false`:

* The user is more likely to run into cyclic dependency issues, due to needing to always alter both the key policy and grantee/principal policy for every permissions update, and because key policies validate principals on creation (creating a dependency from the "key stack" on the "resource stack"). That was the initial motivation for introducing `trustAccountIdentities` (see #5575).
* There is a false sense of better security, in that there is a limited set of actions vs '*'. However, this isn't really any better, as the actions -- by necessity -- includes 'PutKeyPolicy', which allows updating the key policy to allow/enable anything.
* The current permissions set allows unintuitive, asymmetric usage. For example: create a key with `trustAccountIdentities: false` and use it to encrypt a bucket. The root user (or any admin role) in the account can now PUT to the bucket, but not GET.

So that's what I'm trying to avoid. I guess in the "pro" column, with `trustAccountIdentities: false`:

* Minimum permissions - the key is unusable until the key policy is further modified. (e.g., `new Key(this, 'Key')` on its own is unusable by any account or role).

That feels like a really odd set of trade-offs. Maybe there's something else we can target as the alternative behavior, but my thought was at that point, just allowing the user to specify their own policy is likely best.

Reply:
Okay, thanks for the reasoning. So you are saying just raise an error if this is set to `false` (and feature flag is enabled) as well as @deprecate it.

I'd also add a note in the README that mentions how to customize this policy.

Reply:
Yup. The current revision has both the error, and an enhanced README section explaining how to customize the policy.